File: src/main/java/com/smashingmods/alchemistry/client/jei/category/EvaporatorRecipeCategory.java
package com.smashingmods.alchemistry.client.jei.category;

//public class EvaporatorRecipeCategory implements IRecipeCategory<EvaporatorRecipe> {
//    private IGuiHelper guiHelper;
//    private final static int u = 40;
//    private final static int v = 11;
//
//    public EvaporatorRecipeCategory(IGuiHelper guiHelper) {
//        this.guiHelper = guiHelper;
//    }
//
//    @Override
//    public ResourceLocation getUid() {
//        return new ResourceLocation(Alchemistry.MODID, JEIIntegration.EVAPORATOR_CATEGORY);
//    }
//
//    @Override
//    public Class<? extends EvaporatorRecipe> getRecipeClass() {
//        return EvaporatorRecipe.class;
//    }
//
//    @Override
//    public Component getTitle() {
//        return new TextComponent(I18n.get("alchemistry.jei.evaporator"));
//    }
//
//    @Override
//    public IDrawable getBackground() {
//        return guiHelper.createDrawable(new ResourceLocation(Alchemistry.MODID, "textures/gui/evaporator_gui.png"),
//                u, v, 100, 125);
//    }
//
//    @Override
//    public IDrawable getIcon() {
//        return guiHelper.createDrawableIngredient(new ItemStack(BlockRegistry.EVAPORATOR_BLOCK.get()));
//    }
//
//    @Override
//    public void setIngredients(EvaporatorRecipe recipe, IIngredients ingredients) {
//        ingredients.setInput(VanillaTypes.FLUID, recipe.input);
//        ingredients.setOutput(VanillaTypes.ITEM, recipe.output);
//    }
//
//    @Override
//    public void setRecipe(IRecipeLayout recipeLayout, EvaporatorRecipe recipe, IIngredients ingredients) {
//        IGuiItemStackGroup guiItemStacks = recipeLayout.getItemStacks();
//        IGuiFluidStackGroup guiFluidStacks = recipeLayout.getFluidStacks();
//
//
//        int x = 121 - u;
//        int y = 51 - v;
//        guiItemStacks.init(0, false, x, y);
//        guiItemStacks.set(0, ingredients.getOutputs(VanillaTypes.ITEM).get(0));
//
//        x = 48 - u;
//        y = 69 - u;
//        FluidStack inputStack = ingredients.getInputs(VanillaTypes.FLUID).get(0).get(0);
//        if (inputStack.getAmount() > 0) {
//            guiFluidStacks.init(1, true, x, y, 16, 60, inputStack.getAmount(), false, null);
//            guiFluidStacks.set(1, inputStack);
//        }
//    }
//}

File: src/main/kotlin/al132/alchemistry/compat/jei/evaporator/EvaporatorRecipeCategory.kt
package al132.alchemistry.compat.jei.evaporator

import al132.alchemistry.client.GuiEvaporator
import al132.alchemistry.compat.jei.AlchemistryRecipeCategory
import al132.alchemistry.compat.jei.AlchemistryRecipeUID
import al132.alchemistry.compat.jei.Translator
import mezz.jei.api.IGuiHelper
import mezz.jei.api.gui.IRecipeLayout
import mezz.jei.api.ingredients.IIngredients
import net.minecraft.item.ItemStack
import net.minecraftforge.fluids.FluidStack

class EvaporatorRecipeCategory(guiHelper: IGuiHelper)
    : AlchemistryRecipeCategory<EvaporatorRecipeWrapper>(guiHelper.createDrawable(guiTexture, u, v, 100, 125),
        "jei.evaporator.name") {

    override fun getTitle(): String = Translator.translateToLocal("jei.evaporator.name")

    override fun getUid(): String = AlchemistryRecipeUID.EVAPORATOR

    override fun setRecipe(recipeLayout: IRecipeLayout, recipeWrapper: EvaporatorRecipeWrapper, ingredients: IIngredients) {
        val guiItemStacks = recipeLayout.itemStacks
        val guiFluidStacks = recipeLayout.fluidStacks


        var x = 121 - u
        var y = 51 - v
        guiItemStacks.init(OUTPUT_ONE, false, x, y)
        guiItemStacks.set(OUTPUT_ONE, ingredients.getOutputs(ItemStack::class.java)[0])

        x = 48 - u
        y = 69 - u
        val inputStack = ingredients.getInputs(FluidStack::class.java)[0][0]
        guiFluidStacks.init(FLUID_ONE, true, x, y, 16, 60, inputStack.amount, false, null)
        guiFluidStacks.set(FLUID_ONE, inputStack)
    }

    companion object {

        private val OUTPUT_ONE = 1
        private val FLUID_ONE = 1

        private val u = 40
        private val v = 11

        private val guiTexture = GuiEvaporator.textureLocation()
    }
}